### PR TITLE
basic acknowledgement system (#400)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve #400 

You can use any taipy app that takes long to process on the backend to see the effect (table loading, decimator, etc)

### CODE REVIEW

File: gui/src/components/Router.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ import Navigate from "./Taipy/Navigate";
 import Menu from "./Taipy/Menu";
 import GuiDownload from "./Taipy/GuiDownload";
 import ErrorFallback from "../utils/ErrorBoundary";
+import { CircularProgress } from "@mui/material";
```
Comment (FredLL-Avaiga) [inline gui/src/components/Router.tsx]: Prefer the form
`import CircularProgress from "@mui/material/CircularProgress"`
Material UI recommends (for tree shaking I think ?)

And keep the imports grouped :-)

File: gui/src/context/taipyReducers.ts [hunk 2]
```diff
@@ -351,6 +360,9 @@ export const taipyReducer = (state: TaipyState, baseAction: TaipyBaseAction): Ta
             const id = (action as unknown as TaipyIdAction).id;
             storeClientId(id);
             return { ...state, id: id };
+        case Types.Acknowledgement:
+            const ack_id = (action as unknown as TaipyAckAction).id;
```
Comment (FredLL-Avaiga) [inline gui/src/context/taipyReducers.ts]: camel case ?

**Review by FredLL-Avaiga — APPROVED** [review]
I fixed the pandas type issue with a # type ignore comment in my PR

**Review by FabienLelaquais — APPROVED** [review]
Useful! and neat

**Pull request merged**